fix(logging): improve java core's debug logging

padamstx opened the pull request:
This commit adds some debug logging in various
components, and also modifies some tests so that
it's easy to enable debug logging when running
individual sets of tests.

pyrooka commented on src/test/java/com/ibm/cloud/sdk/core/test/service/BaseServiceTest.java:
```diff
@@ -46,7 +49,16 @@
+
+  // Logging level used by this test.
+  // For debugging, set this to Level.FINE or Level.ALL, etc.
+  private static Level logLevel = Level.FINE;
```
```suggestion
  private static Level logLevel = Level.SEVERE;
```
? :)

padamstx replied:
oops

padamstx replied:
fixed

pyrooka reviewed (approved):
LGTM!

dpopp07 commented on src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java:
```diff
@@ -352,7 +351,6 @@ private OkHttpClient reconfigureClientInterceptors(OkHttpClient client, String i
       for (Iterator<Interceptor> iter = builder.interceptors().iterator(); iter.hasNext(); ) {
         Interceptor element = iter.next();
         if (interceptorToRemove.equals(element.getClass().getSimpleName())) {
-          LOG.log(Level.FINE, "Removing interceptor " + element.getClass().getName() + " from http client instance.");
```
I'm curious - why remove these logs?

padamstx replied:
I think I just didn't find those all that useful.  That's my story and I'm sticking to it.   I can easily add them back if needed.

dpopp07 replied:
Sounds good haha, just wanted to understand the justification. I think that's sufficient 🙂

dpopp07 reviewed (approved):
Looks good! 👍

padamstx commented:
Temporarily changed this PR to draft status while I'm working on adding the redact secrets function and simplifying the testcase changes I original made in this PR.

padamstx commented:
@dpopp07 @pyrooka,
I've updated the PR to include a new logging interceptor (HttpLogger) which mimics the behavior of the okhttp3 HttpLoggingInterceptor class, except that it redacts secrets within the debug messages that it logs.

I've also made it easier to enable debug logging...
The easiest way to do this while running tests is like this (in case you wanted to try it out):
```
cd <project-root>
mvn test -Dtest=<testclassname> -Djava.util.logging.config.file=debug-logging.properties

e.g.
mvn test -Dtest=ContainerAuthenticatorTest -Djava.util.logging.config.file=debug-logging.properties
```
(I added debug-logging.properties to the project in this PR).

dpopp07 reviewed (approved):
Updates look good to me 👍

pyrooka reviewed (approved):
Still LGTM!

ibm-devx-sdk commented:
:tada: This PR is included in version 9.21.2 :tada:

The release is available on [GitHub release](https://github.com/IBM/java-sdk-core/releases/tag/9.21.2)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
